internal/provider/storage_volumes_files_data_source.go

```diff
@@ -0,0 +1,239 @@
+type StorageVolumesFilesDataSource struct {
+	config resourceOrDataSourceConfig
+}
+
+// StorageVolumesFileDataSourceModel describes the data source data model.
+type StorageVolumesFileDataSourceModel struct {
+	CxProfileName    types.String `tfsdk:"cx_profile_name"`
+	VolumeName       types.String `tfsdk:"volume_name"`
+	Path             types.String `tfsdk:"path"`
+	BytesUsed        types.Int64  `tfsdk:"bytes_used"`
+	Name             types.String `tfsdk:"name"`
+	OverwriteEnabled types.Bool   `tfsdk:"overwrite_enabled"`
+	Type             types.String `tfsdk:"type"`
+	GroupID          types.Int64  `tfsdk:"group_id"`
+	HardLinksCount   types.Int64  `tfsdk:"hard_links_count"`
+	Size             types.Int64  `tfsdk:"size"`
+	OwnerID          types.Int64  `tfsdk:"owner_id"`
+	InodeNumber      types.Int64  `tfsdk:"inode_number"`
+	IsEmpty          types.Bool   `tfsdk:"is_empty"`
+	Target           types.String `tfsdk:"target"`
+}
+
+// StorageVolumesFilesDataSourceModel describes the data source data model.
+type StorageVolumesFilesDataSourceModel struct {
+	StorageVolumesFiles []StorageVolumesFileDataSourceModel `tfsdk:"storage_volumes_files"`
+	CxProfileName       types.String                        `tfsdk:"cx_profile_name"`
+	VolumeName          types.String                        `tfsdk:"volume_name"`
+	SVMName             types.String                        `tfsdk:"svm_name"`
+	Path                types.String                        `tfsdk:"path"`
+	ByteOffset          types.Int64                         `tfsdk:"byte_offset"`
+	Name                types.String                        `tfsdk:"name"`
+	OverwriteEnabled    types.Bool                          `tfsdk:"overwrite_enabled"`
+}
+
+// Metadata returns the data source type name.
+func (d *StorageVolumesFilesDataSource) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_" + d.config.name
+}
+
+// Schema defines the schema for the data source.
+func (d *StorageVolumesFilesDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "StorageVolumesFiles data source",
+		Attributes: map[string]schema.Attribute{
+			"cx_profile_name": schema.StringAttribute{
+				MarkdownDescription: "Connection profile name",
+				Required:            true,
+			},
+			"volume_name": schema.StringAttribute{
+				MarkdownDescription: "Volume name",
+				Required:            true,
+			},
+			"svm_name": schema.StringAttribute{
+				MarkdownDescription: "svm name",
+				Required:            true,
+			},
+			"path": schema.StringAttribute{
+				MarkdownDescription: "Relative path of a file or directory in the volume",
+				Required:            true,
+			},
+			"byte_offset": schema.Int64Attribute{
+				MarkdownDescription: "The file offset to start reading from",
+				Optional:            true,
```

Comment:
"byte_offset", "overwrite_enabled", "name" can not returned from the API?

Reply:
that's right